thrust/benchmarks/bench/copy/basic.cu

```diff
@@ -52,7 +52,40 @@ static void basic(nvbench::state& state, nvbench::type_list<T>)
   });
 }
 
-using types = nvbench::type_list<nvbench::uint8_t, nvbench::uint16_t, nvbench::uint32_t, nvbench::uint64_t>;
+// Non-trivially-copyable/relocatable type which cannot be copied using std::memcpy or cudaMemcpy
+struct NonTrivial
+{
+  int a;
+  int b;
+
+  _CCCL_HOST_DEVICE NonTrivial()
+  {
+    a = 42;
+    b = 1337;
+  }
+
+  _CCCL_HOST_DEVICE explicit NonTrivial(int i)
+      : a(i)
+      , b(i)
+  {}
+
+  _CCCL_HOST_DEVICE NonTrivial(const NonTrivial& nt)
+      : a(nt.a)
+      , b(nt.b)
+  {}
+
+  _CCCL_HOST_DEVICE auto operator=(const NonTrivial& nt) -> NonTrivial&
```

Comment:
suggestion: my mental model for the value-based vector initialization above is the following:

```cpp
// thrust::device_vector<T> input(elements, T{1});
NonTrivial val(1);
cub::DeviceFor::Bulk(1, [d_vec, val] __device__ (int) { new (d_vec) NonTrivial(val); });
```
The callable is going to be copyed as kernel parameters to the device memory using memcpy. So the assignment operator and copy constructors are not invoked at this point. If we add an assert anywhere in the thrust code that breaks this code, we'll have hard time remembering that we don't actually rely on functional behavior of this code. I'd suggest to add a comment saying that this is needed just to trigger the non-trivial code path.

Reply:
> The callable is going to be copyed as kernel parameters to the device memory using memcpy. 

Wow. What a violation of the type system! :D

I added a comment.